File: android_mobile/src/main/java/com/alexstyl/specialdates/datedetails/NamedaysViewModel.java
package com.alexstyl.specialdates.datedetails;

import com.alexstyl.specialdates.events.namedays.NamesInADate;

final class NamedaysViewModel implements DateDetailsViewModel {

    private final NamesInADate namedays;

    NamedaysViewModel(NamesInADate namedays) {
        this.namedays = namedays;
    }

    public NamesInADate getNamedays() {
        return namedays;
    }

    @Override
    @DateDetailsViewType
    public int getViewType() {
        return DateDetailsViewType.NAMEDAY;
    }
}

File: memento/src/main/java/com/alexstyl/specialdates/events/namedays/activity/NamedaysViewModel.kt
package com.alexstyl.specialdates.events.namedays.activity

data class NamedaysViewModel(val name: String)
    : NamedayScreenViewModel {

    override val viewType: Int
        get() = NamedayScreenViewType.NAMEDAY
    override val id: Long
        get() = hashCode().toLong()


}

